lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -877,11 +881,16 @@ ParseResult FIRParser::parseType(FIRRTLType &result, const Twine &message) {
     break;
   }
 
-  case FIRToken::l_brace_bar:
-    if (parseEnumType(result))
+  case FIRToken::identifier: {
+    StringRef id;
+    if (parseId(id, "expected a type alias name"))
       return failure();
+    auto it = constants.aliasMap.find(id);
+    if (it == constants.aliasMap.end())
+      return failure();
```

Comment:
Should/could this give a more useful message, such as "no type alias '$id' found" or however we phrase such errors elsewhere.  Good candidate for an error test, although tbh probably this PR should be split to add parsing and so on separately? But won't insist on that, however please consider emitting an error here.

It may be good to check the version here as well, although I understand this implicitly will just fail if can't find the type alias (which it can't since can't add one due to the version check there).

Reply:
Also, can/should type aliases support the identifier literal tokens as well?

Reply:
(Or keywords?)

Reply:
Good point, it would be more consistent to align with bundle name rules, but within this PR I'd like to restrict to identifier.